app/src/main/java/com/sun/adsfinder01/ui/home/HomeAdapter.kt

```diff
@@ -0,0 +1,70 @@
+package com.sun.adsfinder01.ui.home
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.adsfinder01.R
+import com.sun.adsfinder01.data.model.Place
+import com.sun.adsfinder01.databinding.ItemPostsBinding
+import java.util.Objects
+
+class HomeAdapter : RecyclerView.Adapter<HomeAdapter.HomeViewHolder>() {
+
+    private var data: List<Place> = ArrayList()
+
+    fun updatePlaces(places: List<Place>) = when {
+        data.isNullOrEmpty() -> {
+            this.data = places
+            notifyItemRangeInserted(0, places.size)
+        }
+        else -> {
+            val result = DiffUtil.calculateDiff(object : DiffUtil.Callback() {
```

Comment:
Đoạn code này dài quá
a nghĩ em nên tạo 1 class riêng cho DiffUtil.Callback cho dễ đọc.
Sample
https://android.jlelse.eu/smart-way-to-update-recyclerview-using-diffutil-345941a160e0